src/Engine/ProtoAssociative/CodeGen.cs

```diff
@@ -3734,6 +3810,124 @@ private AssociativeNode DFSEmitSplitAssign_AST(AssociativeNode node, ref List<As
             return node;
         }
 
+        /// <summary>
+        /// Converts lhs ident lists to a function call
+        /// a.x = 10 
+        ///     -> t = a.%set_x(10)
+        ///     
+        /// a.x.y = b + c 
+        ///     -> a.x.%set_y(b + c)
+        ///     
+        /// a.x[0] = 10
+        ///     ->  tval = a.%get_x()
+        ///         tval[0] = 10         
+        ///         tmp = a.%set_x(tval)
```

Comment:
Maybe we may think about the other way...I know we did this kind of transformation before, but here if `a.x` is a list, there are too many array allocations and copy operation. I'm not even convinced that we should convert it to getter and setter...